src/drivers/mysql/spellbook.js

```diff
@@ -49,7 +49,9 @@ module.exports = {
     const sets = [];
     // Make sure the correct LAST_INSERT_ID is returned.
     // - https://stackoverflow.com/questions/778534/mysql-on-duplicate-key-last-insert-id
-    sets.push(`${escapeId(primaryColumn)} = LAST_INSERT_ID(${escapeId(primaryColumn)})`);
+    if (!columns.includes(primaryColumn)) {
+      sets.push(`${escapeId(primaryColumn)} = LAST_INSERT_ID(${escapeId(primaryColumn)})`);
+    }
```

Comment:
Could you add some description regarding this change?

Reply:
if insert attributes include primary column, `primaryKey = LAST_INSERT_ID(primaryKey)` is not need any more
```js
new Post({ id: 1, title: 'New Post', createdAt: date, updatedAt: date }).upsert().toString()

//  "INSERT INTO `articles` (`id`, `is_private`, `title`, `word_count`, `gmt_create`, `gmt_modified`) 
// VALUES (1, 0, 'New Post', 0, '2017-12-12 00:00:00.000', '2017-12-12 00:00:00.000') 
// ON DUPLICATE KEY UPDATE 
// `id` = LAST_INSERT_ID(`id`), 
// `id`=VALUES(`id`), `is_private`=VALUES(`is_private`), `title`=VALUES(`title`), `word_count`=VALUES(`word_count`), `gmt_modified`=VALUES(`gmt_modified`)"

//  `id` = LAST_INSERT_ID(`id`) is redundant
```